x-pack/legacy/plugins/snapshot_restore/public/app/sections/home/policy_list/policy_details/policy_details.tsx

```diff
@@ -183,53 +189,102 @@ export const PolicyDetails: React.FunctionComponent<Props> = ({
-                      </EuiButton>
-                    );
-                  }}
-                </PolicyExecuteProvider>
-              </EuiFlexItem>
-            </EuiFlexGroup>
+            <PolicyExecuteProvider>
+              {executePolicyPrompt => {
+                return (
+                  <PolicyDeleteProvider>
+                    {deletePolicyPrompt => {
+                      return (
+                        <EuiPopover
+                          id="policyActionMenu"
+                          button={
+                            <EuiButton
+                              data-test-subj="policyActionMenuButton"
+                              iconSide="right"
+                              onClick={() => setIsPopoverOpen(!isPopoverOpen)}
+                              iconType="arrowDown"
+                              fill
+                            >
+                              <FormattedMessage
+                                id="xpack.snapshotRestore.policyDetails.manageButtonLabel"
+                                defaultMessage="Manage policy"
+                              />
+                            </EuiButton>
+                          }
+                          isOpen={isPopoverOpen}
+                          closePopover={() => setIsPopoverOpen(false)}
+                          panelPaddingSize="none"
+                          withTitle
+                          anchorPosition="rightUp"
+                          repositionOnScroll
+                        >
+                          <EuiContextMenu
+                            data-test-subj="policyActionContextMenu"
+                            initialPanelId={0}
+                            panels={[
+                              {
+                                id: 0,
+                                title: i18n.translate(
+                                  'xpack.snapshotRestore.policyDetails.managePanelTitle',
+                                  {
+                                    defaultMessage: 'Policy options',
+                                  }
+                                ),
+                                items: [
+                                  {
+                                    name: i18n.translate(
+                                      'xpack.snapshotRestore.policyDetails.executeButtonLabel',
+                                      {
+                                        defaultMessage: 'Run now',
+                                      }
+                                    ),
+                                    icon: 'play',
+                                    onClick: () => {
+                                      executePolicyPrompt(policyName, () =>
+                                        // Wait a little bit for policy to execute before
+                                        // reloading policy table and policy details
+                                        setTimeout(() => {
```

Comment:
What's the UX intention behind this timeout? Do we want to have similar behavior when hitting `Run` in the table row?

Reply:
Seems like a perfect use case of React `Context` @jen-huang